sixtyfps_runtime/corelib/model.rs

```diff
@@ -618,6 +618,9 @@ impl<C: RepeatedComponent + 'static> Repeater<C> {
 
         let geometry_changed = listview_geometry_tracker
             .evaluate_if_dirty(|| {
+                // Fetch the model again to make sure that it is a dependency of this geometry tracker.
+                let model = self.model().0.expect("invariant: model existence was checked earlier");
```

Comment:
An alternate way would be to mark the geometry tracker explicitly dirty in self.model() when we notice that the model is dirty. I find this solution a bit cleaner though - but happy to change it if desired.

Reply:
I wonder if the invariant should be written in the expect block or in a.comment

Reply:
I'll change it back to a comment block for now, but I like the idea of something more description coming up in the event that the panic happens as we forgot to adjust the code. But we can discuss that separately :)